### PR TITLE
Fix finalorchardroot serialization

### PR DESCRIPTION
**Pull request by sellout**
Previously, `finalorchardroot` and `orchard.anchor` were serialized differently
in the RPC API, which made it difficult to correlate them. This changes the
serialization for `finalorchardroot` to match `orchard.anchor`.

Please ensure this checklist is followed for any pull requests for this repo. This checklist must be checked by both the PR creator and by anyone who reviews the PR.
* [ ] Relevant documentation for this PR has to be completed and reviewed by @mdr0id before the PR can be merged
* [ ] A test plan for the PR must be documented in the PR notes and included in the test plan for the next regular release

As a note, all buildbot tests need to be passing and all appropriate code reviews need to be done before this PR can be merged

### CODE REVIEW

**Review by daira — CHANGES_REQUESTED** [review]
This is an incompatible change. If we decide to do it, then the change would need to be documented in the release notes and RPC help.

**Review by nuttycom — COMMENTED** [review]
I agree with @daira - I think we should do this, but it needs to be mentioned in the release notes. Update `doc/release_notes.md` to describe this change so that it will be automatically included the next time we cut a release.

**Comment by sellout** [comment]
This has been force pushed, folding #6158 into this change, and updating the release_notes and RPC docs.

**Review by nuttycom — APPROVED** [review]
utACK with minor documentation suggestions.

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1247,6 +1252,9 @@ UniValue z_gettreestate(const UniValue& params, bool fHelp)
             "    \"skipHash\": \"hash\",   (string) hash of most recent block with more information\n"
             "    \"commitments\": {\n"
             "      \"finalRoot\": \"hex\", (string)\n"
+            "                          NB: The serialized representation of this field returned by this method\n"
+            "                              was byte-flipped relative to its representation in the `getrawtransaction`\n"
+            "                              output in prior releases up to v5.2.0. This has now been rectified.\n"```
```
Comment (nuttycom) [inline src/rpc/blockchain.cpp:1257]:
```suggestion
            "                              output in prior releases up to v5.2.0. This has now been rectified.\n"
```

**Review by nuttycom — DISMISSED** [review]
utACK

File: qa/rpc-tests/finalorchardroot.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+        new_treestate = self.nodes[0].z_gettreestate(str(-1))
+        assert_equal(new_treestate["sprout"], treestate["sprout"])
+        assert_equal(new_treestate["sapling"], treestate["sapling"])
+        assert new_treestate["orchard"]["commitments"]["finalRoot"] != treestate["orchard"]["commitments"]["finalRoot"]
+        assert new_treestate["orchard"]["commitments"]["finalState"] != treestate["orchard"]["commitments"]["finalState"]
+        assert_equal(len(new_treestate["orchard"]["commitments"]["finalRoot"]), 64)
+        assert_equal(len(new_treestate["orchard"]["commitments"]["finalState"]), 260)
+
+        pass
```
Comment (daira) [inline qa/rpc-tests/finalorchardroot.py:280]:
```suggestion
```

**Review by str4d — APPROVED** [review]
utACK bdde47fb5d6d8dfbdbbc93c4e1168101ee65cbc2

File: src/rpc/blockchain.cpp [hunk 3]
```diff
@@ -716,9 +717,13 @@ UniValue getblock(const UniValue& params, bool fHelp)
...
-            "  \"finalorchardroot\" : \"xxxx\", (string) The root of the Orchard commitment tree after applying this block.\n"
-            "                                        Omitted for blocks prior to NU5 activation. This will be the null\n"
-            "                                        hash if this block has never been connected to a main chain.\n"
+            "  \"finalorchardroot\" : \"xxxx\", (string, optional) The root of the Orchard commitment tree after\n"
+            "                               applying this block. Omitted for blocks prior to NU5 activation. This\n"
+            "                               will be the null hash if this block has never been connected to a\n"
+            "                               main chain.\n"
+            "                               NB: The serialized representation of this field returned by this method\n"
+            "                                   was byte-flipped relative to its representation in the `getrawtransaction`\n"
+            "                                   output in prior releases up to v5.2.0. This has now been rectified.\n"
```
Comment (daira) [inline src/rpc/blockchain.cpp:726]:
```suggestion
            "                                   output in prior releases up to and including v5.2.0. This has now been\n"
            "                                   rectified.\n"
```
("up to" is ambiguous.)

File: src/rpc/blockchain.cpp [hunk 4]
```diff
@@ -1247,6 +1252,9 @@ UniValue z_gettreestate(const UniValue& params, bool fHelp)
             "    \"skipHash\": \"hash\",   (string) hash of most recent block with more information\n"
             "    \"commitments\": {\n"
             "      \"finalRoot\": \"hex\", (string)\n"
+            "                          NB: The serialized representation of this field returned by this method\n"
+            "                              was byte-flipped relative to its representation in the `getrawtransaction`\n"
+            "                              output in prior releases up to v5.2.0. This has now been rectified.\n"
```
Comment (daira) [inline src/rpc/blockchain.cpp:1257]:
```suggestion
            "                              output in prior releases up to v5.2.0. This has now been\n"
            "                              rectified.\n"
```

**Review by daira — APPROVED** [review]
utACK with minor suggestions.

**Review by nuttycom — APPROVED** [review]
utACK https://github.com/zcash/zcash/pull/6152/commits/5514542a0a8e1ad62637fd4d502bfd1cce1ae69a

**Review by daira — COMMENTED** [review]
Post-hoc ACK 5514542a0a8e1ad62637fd4d502bfd1cce1ae69a

**Pull request merged**